twilio/server.js

```diff
@@ -44,13 +46,50 @@ const listener = app.listen(process.env.PORT, function() {
 });
 
 app.post('/', async function(req, res) {
+  const userMsgTimestamp = Date.now().toString(); // Used for Chatbase message timing
   const body = req.body;
   const text = body.Body;
   const id = body.From;
-  const dialogflowResponse = (await sessionClient.detectIntent(
-      text, id, body)).fulfillmentText;
+  const dialogflowResponse = (await sessionClient.detectIntent(text, id, body));
   const twiml = new  MessagingResponse();
-  const message = twiml.message(dialogflowResponse);
+  const message = twiml.message(dialogflowResponse.fulfillmentText);
+  
+  // Enable Chatbase functionality if API key is set & module is loaded
+  if(chatbaseApiKey && chatbase) {
```

Comment:
Could you put this logic into a helper function? This makes the code more readable.